crates/engine/invalid-block-hooks/src/witness.rs

```diff
@@ -364,26 +364,26 @@ where
         Ok(())
     }
 
-    /// Serializes and saves a value to a JSON file in the output directory
-    fn save_file<T: Serialize>(&self, filename: String, value: &T) -> eyre::Result<PathBuf> {
+    fn write_file(&self, filename: String, contents: &[u8]) -> eyre::Result<PathBuf> {
         let path = self.output_directory.join(filename);
-        File::create(&path)?.write_all(serde_json::to_string(value)?.as_bytes())?;
-
+        File::create(&path)?.write_all(contents)?;
```

Comment:
can we map the error here so that the error includes the filepath, like

https://github.com/paradigmxyz/reth/blob/32a80e8c495cb75571a00a1d62eb2f67c0fe7d87/crates/fs-util/src/lib.rs#L50-L57